src/main/java/ua/goit/timonov/enterprise/module_6_2/dao/jdbc/JdbcCookedDishDAO.java

```diff
@@ -52,20 +52,39 @@ public void setJdbcTemplate(JdbcTemplate jdbcTemplate) {
     @Transactional(propagation = Propagation.MANDATORY)
     public void add(int orderId, String dishName, int cookId) {
         if (orderIsClosedByOrderId(orderId)) {
-            String sql = "INSERT INTO cooked_dish VALUES ((SELECT max(cooked_dish.id) FROM cooked_dish) + 1, " +
-                    "(SELECT id FROM dish_to_ordering WHERE order_id = ? AND dish_id = " +
-                    "(SELECT id FROM dish WHERE name = ?)), ?)";
+            String sql = "INSERT INTO cooked_dish VALUES ((SELECT max(cooked_dish.id) FROM cooked_dish) + 1, ?, ?, " +
+                    "(SELECT dish.id FROM dish WHERE dish.name = ?))";
             template.update(sql, orderId, dishName, cookId);
         }
         else {
             throw new IllegalArgumentException("Order is not closed.");
         }
     }
 
+    /**
+     * adds new cooked dish to DB table
+     * @param cookedDish        cooked dish to add
+     * throws                   EmptyResultDataAccessException, DataAccessException
+     */
+    @Override
+    @Transactional(propagation = Propagation.MANDATORY)
```

Comment:
Propagation.MANDATORY supports a current transaction, throws an exception if none exists. Current transaction is opened in the class' CookedDishController method: public List<CookedDish> getAll() with annotation  @Transactional and default value of propagation = Propagation.REQUIRED.